### PR TITLE
Fix track file

### LINKED ISSUE #278: New track_file syntax generates invalid config
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 7.6.1
- Ruby: 2.7.0
- Distribution: Ubuntu 20.04
- Module version: 3.5.0
- Keepalive version: 2.0.19

## How to reproduce (e.g Puppet code you use)

```
  keepalived::vrrp::track_file { 'mysql-active':
    file_name => '/var/run/mysql-active',
    weight    => 0,
    init_file => '1'
  }
```

## What are you seeing

Here's the resulting config in keepalived.conf. I think that `track_file` is not a valid config stanza.

```
track_file mysql-active {
  file     "/var/run/mysql-active"
  weight      0
  init_file 1
}
```

## What behaviour did you expect instead

Should be `vrrp_track_file` instead:

```
vrrp_track_file mysql-active {
  file     "/var/run/mysql-active"
  weight      0
  init_file 1
}
```

## Any additional information you'd like to impart

I can 100% submit a PR for this. Just want to make sure I'm not missing something obvious.

### PR DESCRIPTION
**Pull request by normelton**
#### Pull Request (PR) description
This PR solves two issues related to keepalive configuration:

- For versions of keepalived < 2.1.0, the "track_file" config stanza should be called "vrrp_track_file". The manpage for keepalived.conf confirms the deprecation, I found the exact version by digging through the source code for keepalived.
- The previous implementation of this module allowed a particular file to have a given weight, but when the track_file was referenced inside the vrrp_instance, the weight was reassigned to "1". According to the keepalived.conf manpage, the weight defined inside vrrp_instance defaults to weight configured in track_file. It should be safe to remove.

#### This Pull Request (PR) fixes the following issues
Fixes #278

### CODE REVIEW

**Comment by bastelfreak** [comment]
thanks for the PR/email. Please take a look at the inline comments and the other datatypes you added. If you read the code correctly, this is only a bugfix and shouldn't breaking existing installations?

File: templates/track_file.epp [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<%- | String $name,
+      String $file_name,
```
[the remark this answers is not in the document]
Comment (normelton) [reply templates/track_file.epp:2]: Same here, I'll update vrrp_track_file.epp.

File: templates/track_file.epp [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<%- | String $name,
```
[the remark this answers is not in the document]
Comment (normelton) [reply templates/track_file.epp:1]: I would say the $name should be required. I'll also update the existing vrrp_track_file.epp for consistency.

File: spec/defines/keepalived_vrrp_track_file_spec.rb [hunk 3]
```diff
@@ -11,7 +11,7 @@
   on_supported_os.each do |os, facts|
     context "on #{os}" do
       let(:facts) do
-        facts
+        facts.merge({ 'keepalived_version' => '2.2.0' })
```
Comment (bastelfreak) [inline spec/defines/keepalived_vrrp_track_file_spec.rb:14]: can you extend the tests so they run with keepalived_version 2.2.0 and maybe keepalived_version 2.0.0?

**Comment by normelton** [comment]
Okay, so I realized a corner-case. This patch depends on knowing the value of the `keepalived_version` fact. That's fine, as long as keep alive is installed. But if you're deploying keepalive with this module, then the fact doesn't have a value until the package is installed. The code here will complain that the fact value doesn't exist.

I think I need to withdraw this PR and reconsider. A few possibilities:

- Without a fact value, assume the latest version & syntax. The first "puppet apply" might push out an invalid syntax, which would be resolved on subsequent invocations when the fact is present.
- Abandon the `keepalived_version` fact, and allow the user to specify a version of keepalived in their puppet manifest. Without a version specified, assume the latest. This is less "magical", but works.
- Allow the user to configure either a `vrrp_track_file` or `track_file` in their puppet manifest, and assume they know which syntax is required for their specific version of keepalived.

Thoughts?

**Comment by bluthg** [comment]
Duh, I had my fix lying around for ~ as long as this PR is open... maybe we should join forces? ;-)

**Comment by normelton** [comment]
> Duh, I had my fix (#282) lying around for ~ as long as this PR is open... maybe we should join forces? ;-)

Yeah your solution is much simpler. I tried to make it automagically work, but that gets complicated. I’m closing this PR and upvoting yours.
